r/bootstrap.R

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+if (!dir.exists('../cpp')) {
+  cli::cli_alert_warning("Arrow C++ sources not found, skipping bootstrap.")
+  q(save = "no")
```

Comment:
But that would produce an error if you have a tarball of just /r (potentially with sources already vendored) and the file really isn't there to be `source`d so I don't think that's a valid concern (unlike the `install-arrow.R` case where it also had a standalone functionality).

Reply:
That's actually not true, when I use `devtools::local_install()` with cli_abort it errors (because '.' is copied to a tmp dir that doesn't have cpp)
```
Enter one or more numbers, or an empty line to skip updates: 3
Running bootstrap.R...
Error:
! Arrow C++ sources not found, skipping bootstrap.
Backtrace:
    ▆
 1. └─cli::cli_abort("Arrow C++ sources not found, skipping bootstrap.")
 2.   └─rlang::abort(...)
Execution halted
Error: Failed to install 'arrow' from local:
  ! System command 'Rscript' failed
```